commit 5709157b62e3bc939176b11c95d8094b30f60d05
Author: Isaac Davis <isaac.davis@joyent.com>
Date:   2019-06-10T21:02:37+00:00 (4 months ago)
    
    MANTA-4309 Add registrar config to prometheus image
    TRITON-1723 triton-prometheus uses wrong NODE_PREBUILT_IMAGE

diff --git a/Makefile b/Makefile
index 8e5f6a0..6aa002a 100644
--- a/Makefile
+++ b/Makefile
@@ -14,8 +14,8 @@ GO_PREBUILT_VERSION = 1.11.1
 NODE_PREBUILT_VERSION = v6.15.1
 ifeq ($(shell uname -s),SunOS)
     NODE_PREBUILT_TAG=zone
-    # Allow building on other than sdc-minimal-multiarch-lts@15.4.1
-    NODE_PREBUILT_IMAGE=18b094b0-eb01-11e5-80c1-175dac7ddf02
+    # Allow building on other than triton-origin-multiarch-18.1.0
+    NODE_PREBUILT_IMAGE=1ad363ec-3b83-11e8-8521-2f68a4a34d5d
 endif
 
 ENGBLD_USE_BUILDIMAGE = true
diff --git a/sapi_manifests/registrar/manifest.json b/sapi_manifests/registrar/manifest.json
new file mode 100644
index 0000000..6c99cdc
--- /dev/null
+++ b/sapi_manifests/registrar/manifest.json
@@ -0,0 +1,4 @@
+{
+    "name": "registrar",
+    "path": "/opt/smartdc/registrar/etc/config.json"
+}
\ No newline at end of file
diff --git a/sapi_manifests/registrar/template b/sapi_manifests/registrar/template
new file mode 100644
index 0000000..8e6d38b
--- /dev/null
+++ b/sapi_manifests/registrar/template
@@ -0,0 +1,40 @@
+{
+    "registration": {
+        {{#is_manta_service}}
+            "domain": "{{SERVICE_NAME}}",
+        {{/is_manta_service}}
+        {{^is_manta_service}}
+            "domain": "{{SERVICE_NAME}}.{{datacenter_name}}.{{dns_domain}}",
+        {{/is_manta_service}}
+        "type": "load_balancer",
+        "service": {
+            "type": "service",
+            "service": {
+                "srvce": "_prometheus",
+                "proto": "_tcp",
+                "port": 9090
+            },
+            "ttl": 60
+        },
+        "ttl": 30
+    },
+
+    "zookeeper": {
+        "servers": [
+            {{#ZK_SERVERS}}
+                {
+                    "host": "{{host}}",
+                    "port": {{port}}
+                }{{^last}}, {{/last}}
+            {{/ZK_SERVERS}}
+        ],
+        "timeout": 60000
+    },
+
+    {{#is_manta_service}}
+        "adminIp": "{{auto.MANTA_IP}}"
+    {{/is_manta_service}}
+    {{^is_manta_service}}
+        "adminIp": "{{auto.ADMIN_IP}}"
+    {{/is_manta_service}}
+}
